test/common/http/date_provider_impl_test.cc

```diff
@@ -11,11 +11,63 @@
 #include "gtest/gtest.h"
 
 using testing::NiceMock;
+using testing::Return;
 
 namespace Envoy {
 namespace Http {
 
-TEST(DateProviderImplTest, All) {
+class MockTimeSource : public TimeSource {
+public:
+  MOCK_METHOD(SystemTime, systemTime, ());
+  MOCK_METHOD(MonotonicTime, monotonicTime, ());
+};
+
+TEST(SlowDateProviderImplTest, RequestHeaders) {
+  MockTimeSource time_source;
+  SystemTime time_point{SystemTime::duration(2000000)};
+  EXPECT_CALL(time_source, systemTime()).WillOnce(Return(time_point));
+
+  SlowDateProviderImpl provider(time_source);
+  TestRequestHeaderMapImpl headers;
+  EXPECT_EQ(nullptr, headers.Date());
+
+  provider.setDateHeader(headers);
+  EXPECT_EQ(headers.get_("date"), "Thu, 01 Jan 1970 00:00:02 GMT");
+}
+
+TEST(SlowDateProviderImplTest, ResponseHeaders) {
+  MockTimeSource time_source;
+  SystemTime time_point{SystemTime::duration(2000000)};
+  EXPECT_CALL(time_source, systemTime()).WillOnce(Return(time_point));
+
+  SlowDateProviderImpl provider(time_source);
+  TestResponseHeaderMapImpl headers;
+  EXPECT_EQ(nullptr, headers.Date());
+
+  provider.setDateHeader(headers);
+  EXPECT_EQ(headers.Date()->value(), "Thu, 01 Jan 1970 00:00:02 GMT");
+}
+
+TEST(TlsCachingDateProviderImplTest, TlsRequestHeaders) {
```

Comment:
This test is copy/pasted from the response test below

Reply:
Thanks for adding the slow date provider tests as well.